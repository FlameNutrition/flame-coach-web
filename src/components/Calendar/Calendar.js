import React, { useState } from 'react';
import clsx from 'clsx';
import PropTypes from 'prop-types';
import { NavigateNextRounded as NextMonth, NavigateBeforeRounded as PreviousMonth } from '@material-ui/icons';
import {
  Box, Card, CardContent, CardHeader, Divider, makeStyles, Typography
} from '@material-ui/core';
import CalendarReact from 'react-calendar';
// FIXME: Try to understand why the Calendar.module.css does not work
import './Calendar.css';

const useStyles = makeStyles(() => ({
  root: {
    height: '100%'
  }
}));

const Calendar = ({
  daily, onChangeCalendar
}) => {
  const classes = useStyles();
  const [calendar, setCalendar] = useState(null);

  const infoDaily = (
    <Typography component="div" align="right" variant="caption">
      <strong>Day:</strong>
      {' '}
      Double click date
    </Typography>
  );

  const infoPeriod = (
    <>
      {infoDaily}
      <Typography component="div" align="right" variant="caption">
        <strong>Period:</strong>
        {' '}
        Select a start and end date
      </Typography>
    </>
  );

  return (
    <Card
      className={clsx(classes.root)}
    >
      <CardHeader title="Calendar" />
      <Divider />
      <CardContent>
        <Box>
          <CalendarReact
            onChange={(value) => {
              setCalendar(value);
              onChangeCalendar(value);
            }}
            selectRange={!daily}
            value={calendar}
            nextLabel={<NextMonth />}
            next2Label={null}
            returnValue="range"
            prevLabel={<PreviousMonth />}
            prev2Label={null}
          />
        </Box>
        <Box>
          {!daily ? infoPeriod : null}
        </Box>
      </CardContent>
    </Card>
  );
};

Calendar.propTypes = {
  daily: PropTypes.bool,
  onChangeCalendar: PropTypes.func.isRequired
};

Calendar.defaultProps = {
  daily: false,
};
export default Calendar;
