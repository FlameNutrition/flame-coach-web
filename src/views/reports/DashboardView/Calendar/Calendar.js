import React, { useState } from 'react';
import clsx from 'clsx';
import logger from 'loglevel';
import PropTypes from 'prop-types';
import moment from 'moment';
import NextMonth from '@material-ui/icons/NavigateNextRounded';
import PreviousMonth from '@material-ui/icons/NavigateBeforeRounded';
import {
  Box, Card, CardContent, CardHeader, Divider, makeStyles
} from '@material-ui/core';
import Calendar from 'react-calendar';
// FIXME: Try to understand why the Calendar.module.css does not work
import './Calendar.css';

const useStyles = makeStyles(() => ({
  root: {
    height: '100%'
  }
}));

const TrafficByDevice = ({ className, ...rest }) => {
  const classes = useStyles();
  const [calendar, setCalendar] = useState(new Date());

  const handleChangeCalendar = (value) => {
    logger.info('Change the calendar date', moment(value));
    setCalendar(value);
  };

  return (
    <Card
      className={clsx(classes.root, className)}
      {...rest}
    >
      <CardHeader title="Calendar" />
      <Divider />
      <CardContent>
        <Box
          position="relative"
        >
          <Calendar
            onChange={handleChangeCalendar}
            value={calendar}
            nextLabel={<NextMonth />}
            next2Label={null}
            prevLabel={<PreviousMonth />}
            prev2Label={null}
          />
        </Box>
      </CardContent>
    </Card>
  );
};

TrafficByDevice.propTypes = {
  className: PropTypes.string
};

export default TrafficByDevice;
