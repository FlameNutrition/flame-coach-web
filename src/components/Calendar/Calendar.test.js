import React from 'react';
import toJson from 'enzyme-to-json';
import { createShallow } from '@material-ui/core/test-utils';
import sinon from 'sinon';
import Calendar from './Calendar';

describe('<Calendar/ >', () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  it('Show daily selection calendar', () => {
    const changeCalenderHandler = sinon.spy();

    const calendar = shallow(
      <Calendar
        daily
        onChangeCalendar={changeCalenderHandler}
      />
    );

    expect(toJson(calendar)).toMatchSnapshot();
  });

  it('Show multiple selection calendar', () => {
    const changeCalenderHandler = sinon.spy();

    const calendar = shallow(
      <Calendar onChangeCalendar={changeCalenderHandler} />
    );

    expect(toJson(calendar)).toMatchSnapshot();

    const labelBox = calendar.find('WithStyles(ForwardRef(Typography))');

    console.log(labelBox.debug());
  });
});
