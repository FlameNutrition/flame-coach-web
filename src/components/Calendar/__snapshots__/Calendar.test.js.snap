// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Calendar/ > Show daily calendar 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardHeader))
    title="Calendar"
  />
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(CardContent))>
    <Styled(MuiBox)>
      <Calendar
        maxDate={+275760-09-13T00:00:00.000Z}
        maxDetail="month"
        minDate={0001-01-01T00:00:00.000Z}
        minDetail="century"
        next2Label={null}
        nextLabel={<Memo />}
        onChange={[Function]}
        prev2Label={null}
        prevLabel={<Memo />}
        returnValue="range"
        selectRange={false}
        showNavigation={true}
        showNeighboringMonth={true}
        value={null}
      />
    </Styled(MuiBox)>
    <Styled(MuiBox) />
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`<Calendar/ > Show daily selection calendar 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardHeader))
    title="Calendar"
  />
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(CardContent))>
    <Styled(MuiBox)>
      <Calendar
        maxDate={+275760-09-13T00:00:00.000Z}
        maxDetail="month"
        minDate={0001-01-01T00:00:00.000Z}
        minDetail="century"
        next2Label={null}
        nextLabel={<Memo />}
        onChange={[Function]}
        prev2Label={null}
        prevLabel={<Memo />}
        returnValue="range"
        selectRange={false}
        showNavigation={true}
        showNeighboringMonth={true}
        value={null}
      />
    </Styled(MuiBox)>
    <Styled(MuiBox) />
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;

exports[`<Calendar/ > Show multiple selection calendar 1`] = `
<WithStyles(ForwardRef(Card))
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(CardHeader))
    title="Calendar"
  />
  <WithStyles(ForwardRef(Divider)) />
  <WithStyles(ForwardRef(CardContent))>
    <Styled(MuiBox)>
      <Calendar
        maxDate={+275760-09-13T00:00:00.000Z}
        maxDetail="month"
        minDate={0001-01-01T00:00:00.000Z}
        minDetail="century"
        next2Label={null}
        nextLabel={<Memo />}
        onChange={[Function]}
        prev2Label={null}
        prevLabel={<Memo />}
        returnValue="range"
        selectRange={true}
        showNavigation={true}
        showNeighboringMonth={true}
        value={null}
      />
    </Styled(MuiBox)>
    <Styled(MuiBox)>
      <WithStyles(ForwardRef(Typography))
        align="right"
        component="div"
        variant="caption"
      >
        <strong>
          Day:
        </strong>
         
        Double click date
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Typography))
        align="right"
        component="div"
        variant="caption"
      >
        <strong>
          Period:
        </strong>
         
        Select a start and end date
      </WithStyles(ForwardRef(Typography))>
    </Styled(MuiBox)>
  </WithStyles(ForwardRef(CardContent))>
</WithStyles(ForwardRef(Card))>
`;
